modules/azure/storage.go

```diff
@@ -0,0 +1,246 @@
+	storageAccount, err := GetStorageAccountPropertyE(storageAccountName, resourceGroupName, subscriptionID)
+	if err != nil {
+		return "", err
+	}
+	return string(storageAccount.Sku.Tier), nil
+}
+
+// GetStorageBlobContainerE returns Blob container client.
+// This function would fail the test if there is an error.
+func GetStorageBlobContainerE(containerName, storageAccountName, resourceGroupName, subscriptionID string) (*storage.BlobContainer, error) {
+	subscriptionID, err := getTargetAzureSubscription(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+	resourceGroupName, err2 := getTargetAzureResourceGroupName((resourceGroupName))
+	if err2 != nil {
+		return nil, err2
+	}
+	client, err := GetStorageBlobContainerClientE(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+	container, err := client.Get(context.Background(), resourceGroupName, storageAccountName, containerName)
+	if err != nil {
+		return nil, err
+	}
+	return &container, nil
+}
+
+// GetStorageAccountPropertyE returns StorageAccount properties.
+// This function would fail the test if there is an error.
+func GetStorageAccountPropertyE(storageAccountName, resourceGroupName, subscriptionID string) (*storage.Account, error) {
+	subscriptionID, err := getTargetAzureSubscription(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+	resourceGroupName, err2 := getTargetAzureResourceGroupName((resourceGroupName))
+	if err2 != nil {
+		return nil, err2
+	}
+	client, err := GetStorageAccountClientE(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+	account, err := client.GetProperties(context.Background(), resourceGroupName, storageAccountName, "")
+	if err != nil {
+		return nil, err
+	}
+	return &account, nil
+}
+
+// GetStorageAccountClientE creates a storage account client.
+// This function would fail the test if there is an error.
+func GetStorageAccountClientE(subscriptionID string) (*storage.AccountsClient, error) {
+	storageAccountClient := storage.NewAccountsClient(os.Getenv(AzureSubscriptionID))
+	authorizer, err := NewAuthorizer()
+	if err != nil {
+		return nil, err
+	}
+	storageAccountClient.Authorizer = *authorizer
+	return &storageAccountClient, nil
+}
+
+// GetStorageBlobContainerClientE creates a storage container client.
+// This function would fail the test if there is an error.
+func GetStorageBlobContainerClientE(subscriptionID string) (*storage.BlobContainersClient, error) {
+	blobContainerClient := storage.NewBlobContainersClient(os.Getenv(AzureSubscriptionID))
+	authorizer, err := NewAuthorizer()
+
+	if err != nil {
+		return nil, err
+	}
+	blobContainerClient.Authorizer = *authorizer
+	return &blobContainerClient, nil
+}
+
+// GetStorageURISuffixE returns the proper storage URI suffix for the configured Azure environment.
+// This function would fail the test if there is an error.
+func GetStorageURISuffixE() (*string, error) {
+	var envName = "AzurePublicCloud"
```

Comment:
Need @mjyeaney to weight in on this with sovereign cloud PR coming.

Reply:
This will get replaced / updated by the `client_factory` methods that handle looking up the Azure env name (for Public, USGov, China, Germany, or Stack). This is likely fine until then.